### ISSUE TITLE
Warning "Try using .loc[row_indexer,col_indexer] = value instead" is given when using ".loc"

### ISSUE THREAD

**Issue by mborus** [pandas-dev/pandas#33237]
#### Code Sample, a copy-pastable example if possible

TLDR; Pandas warning is confusing since it recommends what seems to cause the warning.

```python

    # run this in one cell of a freshly started Jupyterlab Notebook
    # the warning is shown once, to see it again, restart the kernel

    import pandas as pd
    df = pd.DataFrame([list(range(1000)), list(range(1000)), list(range(1000)), list(range(1000))]).T
    df2 = df[[1, 3]]
    df2.loc[df2[1] > 500, 'marker'] = True
   

```
#### Problem description

This gives out the following warning:

```
C:\Python37\lib\site-packages\pandas\core\indexing.py:844: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self.obj[key] = _infer_fill_value(value)
C:\Python37\lib\site-packages\pandas\core\indexing.py:965: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self.obj[item] = s

```
This warning has confused me a many, many times, mainly, because it suggests
"try using .loc" when that's exactly what I'm doing.

(I followed the link and read the doc and now know that I should have made a copy the
line before) - still the suggestion to use .loc is not helful here. 

#### Expected Output

Remove the line "Try using .loc[row_indexer,col_indexer] = value instead"
when the code is using ".loc". The rest of the warning is fine.

#### Output of ``pd.show_versions()``

<details>

[paste the output of ``pd.show_versions()`` here below this line]
INSTALLED VERSIONS
------------------
commit           : None
python           : 3.7.4.final.0
python-bits      : 64
OS               : Windows
OS-release       : 7
machine          : AMD64
processor        : Intel64 Family 6 Model 42 Stepping 7, GenuineIntel
byteorder        : little
LC_ALL           : None
LANG             : DE
LOCALE           : None.None

pandas           : 1.0.1
numpy            : 1.16.2
pytz             : 2018.7
dateutil         : 2.7.5
pip              : 20.0.2
setuptools       : 40.8.0
Cython           : None
pytest           : 5.2.0
hypothesis       : None
sphinx           : 1.7.5
blosc            : None
feather          : None
xlsxwriter       : 1.1.2
lxml.etree       : 4.3.2
html5lib         : None
pymysql          : 0.9.3
psycopg2         : None
jinja2           : 2.11.1
IPython          : 7.0.1
pandas_datareader: None
bs4              : 4.6.3
bottleneck       : None
fastparquet      : None
gcsfs            : None
lxml.etree       : 4.3.2
matplotlib       : 3.0.3
numexpr          : None
odfpy            : None
openpyxl         : 2.5.12
pandas_gbq       : None
pyarrow          : None
pytables         : None
pytest           : 5.2.0
pyxlsb           : None
s3fs             : None
scipy            : 1.2.1
sqlalchemy       : 1.2.15
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
xlsxwriter       : 1.1.2
numba            : None

</details>

**Comment by TomAugspurger** [comment]
Can you provide a reproducible example? http://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports

**Comment by mborus** [comment]
Updated with reproducable example

**Comment by jreback** [comment]
this is as expected df2 is creating. view onto df which u r then modifying 

simply copy df2

**Comment by TomAugspurger** [comment]
Thanks. The warning is correct there, since it's not clear if `df` will / should be modified along with df2.
